share/doc/homebrew/How-to-Create-and-Maintain-a-Tap.md

```diff
@@ -0,0 +1,42 @@
+# How to Create and Maintain a Tap
+Taps are external sources of Homebrew formulae. They can be created by anyone
+to provide their own formulae to any Homebrew user.
+
+## Creating a tap
+A tap must be a git repository available online. It can be hosted on GitHub or
+elsewhere. If hosted on GitHub, the repo’s name must start with `homebrew-`.
+
+Formulae follow the same format as the core’s ones, and can be added under
+`Formula` or `HomebrewFormula`. It works if they are at the root of the repo
+but this is not recommended.
```

Comment:
The reasons are:
* We treat `Formula` folder with highest priority. [Related code](https://github.com/Homebrew/homebrew/blob/master/Library/Homebrew/formulary.rb#L248-L250)
* It's more organize considering with other folders such as `cmd` and `Casks` which have their own usage.
* It will offer a better view on Github page. For example, it's very hard to see the README for `homebrew/science` since there are so many formulae in the root of the repo.